ui/app/components/clients/attribution.js

```diff
@@ -103,9 +104,24 @@ export default class Attribution extends Component {
     }
   }
 
-  get getCsvData() {
+  destructureCountsToArray(object) {
+    let { clients, entity_clients, non_entity_clients } = object;
+    return [clients, entity_clients, non_entity_clients];
+  }
+
+  constructCsvRow(firstColumn, secondColumn = null, otherColumns) {
+    // ['ns label', 'mount label', 'total clients', 'entity', 'non-entity']
+    // if constructing namespace row, mount column will be blank
+    // if firstColumn is a string, then we're at mount level attribution
```

Comment:
NIT: but is it an array if it's not a string? Maybe add to comment.

Reply:
Again this is where typescript helps a user read the code. i can tell from the firstColumn.label below it's an object, but might help a reader understand what's expected if the type is called out.  (maybe a larger discussion to have as a team about calling out types in comments since we don't typescript).

Reply:
`firstColumn` is a string if we're within a namespace, but an object if we're iterating through a namespace. I had abstracted to be `firstColumn` and `secondColumn` - but as this is the only place this is used right now, changed the variable names for clarity. 

I definitely want the comment to be clear! Is this better? 

```
  constructCsvRow(namespaceColumn, mountColumn = null, otherColumns) {
    // returns an array that corresponds to a row in the csv file:
    // ['ns label', 'mount label', total client #, entity #, non-entity #]
    // if namespaceColumn is a string, then we're at mount level attribution, otherwise it is an object
    // if constructing a namespace row, the mountColumn=null so the column is blank, otherwise it is an object
    return [
      `${typeof namespaceColumn === 'string' ? namespaceColumn : namespaceColumn.label}`,
      `${mountColumn ? mountColumn.label : ''}`,
      ...otherColumns,
    ];
  }
```